### PR TITLE
[Bugfix:HelpQueue] Allow empty queue codes and add tests

### LINKED ISSUE #10895: Office Hours Queue -- cannot set queue access code to empty string or NULL
**Describe the bug**
An Office Hours Queue cannot be configured with an empty string or NULL value.
A queue code seems to be required now. 

**Expected behavior**
The queue access code should be optional.

UPDATE:  bug fix in PR #10901 - Issue will remain open until Cypress tests are added in future PR.

### PR DESCRIPTION
**Pull request by lavalleeale**
### Please check if the PR fulfills these requirements:

* [X] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
Attempting to modify a queue with an empty queue code will both fail at the html form validation level and at the server level.
Fixes #10895

### What is the new behavior?
Modifying a queue with an empty code will work as expected.

### Other information?
Tests have been updated to test both creation and modification of blank queue codes

### CODE REVIEW

**Review by williamschen23 — CHANGES_REQUESTED** [review]
Can you also make the student join and leave the blank queue? Also @IDzyre thoughts on changing the redirection limit here.

The other aspects of the PR works as intended.

**Comment by IDzyre** [comment]
> Can you also make the student join and leave the blank queue? Also @IDzyre thoughts on changing the redirection limit here.
> 
> The other aspects of the PR works as intended.

If it is absolutely needed then I think it's okay, however if you can find a way to not then that would be ideal, as more redirects cause more time.

**Comment by lavalleeale** [comment]
Yeah, I removed the line by splitting up the test, I think there might be a glitch in Cypress here where it's not that a single request has that many redirects, but since the test keeps redirecting from (or too I honestly don't know) the same url the count adds up over the course of the test.

**Review by williamschen23 — CHANGES_REQUESTED** [review]

File: site/cypress/e2e/Cypress-Feature/office_hours_queue.spec.js [hunk 1]
```diff
@@ -88,27 +93,39 @@ describe('test office hours queue', () => {
...
         changeQueueCode(queueName, newQueueCode);
         cy.get('[data-testid="popup-message"]').should('contain', 'Queue Access Code Changed');
 
         openNewQueue(queueName1, queueCode1);
-
+    });
+    it('Joining queues as student', () => {
         // switch to student to join queue
-        switchUser('student');
+        cy.login('student');
```
[the remark this answers is not in the document]
Comment (williamschen23) [reply site/cypress/e2e/Cypress-Feature/office_hours_queue.spec.js:111]: otherwise the PR looks good!

**Review by williamschen23 — APPROVED** [review]

**Review by williamjallen — APPROVED** [review]
Code looks good to me

**Pull request merged**